flake.nix

```diff
@@ -0,0 +1,37 @@
+{
+  description = "A Nix-flake-based Zig development environment";
+
+  inputs.nixpkgs.url = "https://flakehub.com/f/NixOS/nixpkgs/0.1.*.tar.gz";
+
+  outputs = {
+    self,
+    nixpkgs,
+  }: let
+    supportedSystems = ["x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin"];
+    forEachSupportedSystem = f:
+      nixpkgs.lib.genAttrs supportedSystems (system:
+        f {
+          pkgs = import nixpkgs {inherit system;};
+        });
+  in {
+    devShells = forEachSupportedSystem ({pkgs}: {
+      default = pkgs.mkShell {
+        packages = with pkgs; [
+          zig
+          zls
```

Comment:
Are these the right version? You might wanna use [zig-overlay](https://github.com/mitchellh/zig-overlay) here and pin the version

Reply:
Added the zig overlay.